test/middleware-test.js

```diff
@@ -0,0 +1,165 @@
+'use strict';
+
+var expect = require('chai').expect;
+var RSVP = require('rsvp');
+var broccoliMiddleware = require('./../lib/middleware');
+var fixture = require('./helpers/fixture-path');
+var TestHTTPServer = require('./helpers/test-http-server');
+
+describe('broccoli-middleware', function() {
+  describe('watcher resolves correctly', function() {
+    var watcher;
+    var server;
+
+    beforeEach(function() {
+      watcher = RSVP.Promise.resolve();
+
+      watcher['builder'] = {};
+    });
+
+    afterEach(function() {
+      server.stop();
+      server = null;
+    })
+
+    it('responds with the given file if file is on disk', function() {
+      watcher['builder']['outputPath'] = fixture('basic-file');
+      var middleware = broccoliMiddleware(watcher, {
+        autoIndex: false
+      });
+
+      server = new TestHTTPServer(middleware);
+
+      return server.start()
+        .then(function(info) {
+          return server.request('/index.html', {
+            info: info
+          });
+        })
+        .then(function(content) {
+          expect(content).to.match(/This is broccoli middleware page/);
+        });
+    });
+
+    it('responds with error if file not found', function(done) {
```

Comment:
@Turbo87 : That's interesting. I am surprised how CI didn't catch it earlier unless mocha added that assertion in a later version. 

I'll debug and see if I can fix it.